src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,257 @@
+package com.mate.dao.jdbcimpl;
+
+import com.mate.dao.CarDao;
+import com.mate.exception.DataProcessingException;
+import com.mate.lib.Dao;
+import com.mate.model.Car;
+import com.mate.model.Driver;
+import com.mate.model.Manufacturer;
+import com.mate.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertQuery = "INSERT INTO cars (model, manufacturer_id)"
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                             insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            insertAllDrivers(car.getId(), car.getDrivers());
```

Comment:
here you open one connection inside another - let's move this line below the catch block